### DISCUSSION TITLE
Direct Paperless-ngx to save media and data directories to SMB share

### DISCUSSION THREAD

**Discussion by aslm1** [tteck/Proxmox#1179] [Help Needed]
I am trying to have Paperless-ngx use an SMB share for its media and data directories.

I managed to install `cifs-utils` and edited `fstab` so that the share I am trying to direct to is mounted when the LXC starts as: `/mnt/documents/`

In `/mnt/documents/` there are now 2 subdirectories, `data` and `media`.

The part of my `/opt/paperless/paperless.conf` that I changed looks like this:

```
# Paths and folders

#PAPERLESS_CONSUMPTION_DIR=../consume
#PAPERLESS_DATA_DIR=/mnt/documents/data
#PAPERLESS_TRASH_DIR=
#PAPERLESS_MEDIA_ROOT=/mnt/documents/media
#PAPERLESS_STATICDIR=../static
#PAPERLESS_FILENAME_FORMAT=
#PAPERLESS_FILENAME_FORMAT_REMOVE_NONE=
```
After restarting the LXC, consumed documents continue to go to `/opt/paperless/media/documents/originals/` and `/opt/paperless/media/documents/archive/`.

The log files that I have been able to find have been unhelpful.

Any idea what I'm missing?

**Comment by djsayles** [comment]
You need to remove the # symbol from the beginning of the lines that you modified in order for them to be read.

**Reply by aslm1** [reply]
Oh wow... I feel silly that I didn't catch that... thanks for your help with this.

After a restart the media directory on the SMB share is working as expected.

Unfortunately, if I direct the `data` directory to the relevant mounted folder - when I try to login to Paperless-ngx I get a `Server Error (500)`.  Interestingly, files are being created in the `data `directory.  Not quite sure where to go from here, but at least I don't have to worry about running out of space in the LXC container.

**Reply by djsayles** [reply]
I had the same issue with the data directory so I did the same as you.

**Reply by JueFri** [reply]
I am also trying to have Paperless-ngx use a SMB share for its consume, media and data directories. When i change the Paths & folders in the paperless.conf file to the new SMB share folderes and restart paperless and  try to login to Paperless-ngx i get a **Server Error (500)**. Does anybody have some hints or some add. information to solve this ?

**Reply by djsayles** [reply]
As above really.  I could redirect "consume" and it worked fine.  If I redirected "data" I got the 500 error.  So I simply left "data" alone :)

**Reply by dsanner** [reply]
i'm getting the same error. i tried moving old contents of data to smb share data dir and did not work either

**Comment by JueFri** [comment]
Thank you for your very quick reply and your recommendation. I followed your recommendation and i am able to login. Great. But i would like to store the "data" on a NAS. Maybe there is someone who has an idea why the "data" directory cannot be moved to a SMB share.

**Reply by FelixArndt** [reply]
Exact same issue here. Would like to store all documents/data on a NAS. Somebody got it working with a share?

**Comment by Smile2Grow** [comment]
I have found an alternative way to install an SMB. It is also possible to install Webmin via the bash script `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/webmin.sh)"
.` Under "Un-used modules", "Samba Windows File Sharing" can be installed and configured via the web gui. Thank you @tteck !

**Reply by TomHilde** [reply]
> Under "Un-used modules", "Samba Windows File Sharing" can be installed and configured via the web gui.<

Can you explain that to me?
I have installed Webmin.... and now?
